### PR TITLE
Update from voxpupuli modulesync_config

### PR DESCRIPTION
**Pull request by jyaworski**

### CODE REVIEW

**Comment by jyaworski** [comment]
This may take a while.

**Comment by jyaworski** [comment]
This will require a significant refactor of the tests.

**Comment by jyaworski** [comment]
Calling for help on this one. @bastelfreak @nibalizer ?

**Comment by juniorsysadmin** [comment]
As master is at the moment, the critical gems used when testing against Ruby 1.9.3 are: puppet 4.3.2, facter-2.4.6, rspec 2.10.0 and mocha 0.10.5 .

I would pin these in the Gemfile and move your way up slowly taking note of deprecation warnings - rspec 2.10.0 and mocha 0.10.5 breaking changes are mostly like the culprit

File: lib/puppetx/filemapper.rb [hunk 1]
```diff
@@ -189,9 +180,8 @@ def load_all_providers_from_disk
     #
     # @param [Hash<String, Puppet::Resource>] resources
     def prefetch(resources = {})
-
       # generate hash of {provider_name => provider}
-      providers = instances.inject({}) do |hash, instance|
+      providers = instances.each_with_object({}) do |instance, hash|
```
Comment (juniorsysadmin) [inline lib/puppetx/filemapper.rb]: each_with_object is not available in Ruby 1.8.7, but you probably knew that already

Comment (jyaworski) [inline lib/puppetx/filemapper.rb]: Yes. We actually dropped Ruby 1.8.7 support. There should likely be a changelog entry.

**Comment by jyaworski** [comment]
Tag @juniorsysadmin for review. I think this is good to merge; we will likely need at least some scream tests due to the amount of work this module had previously (not much)....

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,17 @@
 CHANGELOG
 =========
 
+1.2.0
```
Comment (juniorsysadmin) [inline CHANGELOG]: This used to test against Ruby 1.8.7 but we never said it was supported in the docs, so I am fine with this versioning.

**Comment by juniorsysadmin** [comment]
Looks reasonable to me

**Comment by juniorsysadmin** [comment]
This module of course needs a Travis token in in order for releases to be pushed, so see the appropriate people about this.

**Comment by jyaworski** [comment]
Tagging @nibalizer for the secret.

**Comment by jyaworski** [comment]
@juniorsysadmin @nibalizer just got me a secret, so it looks good now.

**Pull request merged**